### ISSUE TITLE
Just one mock with "preserve" doesn't work

### ISSUE THREAD

**Issue by naag** [archan937/mecks_unit#2]
Hi all,

first of all, thanks for this great hex, it's looking really awesome! :-)

We noticed that the new `preserve: true` feature doesn't work if there's just one Mock defined in the test, for example:

```
defmodule Foo
  def bar(x), do: x
end

defmodule FooTest do
  use ExUnit.Case, async: true
  use MecksUnit.Case

  # defmock Foo do
  #   def bar(_x), do: 1
  # end

  defmock Foo, preserve: true do
    def bar(_x), do: 1
  end

  mocked_test "this should work" do
    assert Foo.bar(nil) == 1
  end

  mocked_test "this should work as well" do
    assert Foo.bar(nil) == 1
  end
end
```

```
mix test test/foo_test.exs 

FooTest
  * test this should work as well (2.9ms)

  1) test this should work as well (FooTest)
     test/foo_test.exs:17
     Assertion with == failed
     code:  assert Foo.bar(nil) == 1
     left:  nil
     right: 1
     stacktrace:
       test/foo_test.exs:18: (test)

  * test this should work (0.02ms)

  2) test this should work (FooTest)
     test/foo_test.exs:13
     Assertion with == failed
     code:  assert Foo.bar(nil) == 1
     left:  nil
     right: 1
     stacktrace:
       test/foo_test.exs:14: (test)



Finished in 0.04 seconds
2 tests, 2 failures

Randomized with seed 711881
```

By activating the first mock, the second one actually starts working.

**Comment by archan937** [comment]
Hey, @naag. First of all, I'm glad to hear you like the package.

I will look into this issue upcoming weekend. Stay tuned :)

**Comment by naag** [comment]
Thanks for looking into this :-)

**Comment by archan937** [comment]
Sorry for the long wait but I managed to solve the issue. See commit: https://github.com/archan937/mecks_unit/commit/12d344f46c1898ba56fca2c65730af6733c25c8c

I just released MecksUnit v0.1.8. See its CHANGELOG: https://github.com/archan937/mecks_unit/commit/2da6445c8ad95d4913a734fd849b141ca46d1a6b

Good luck with MecksUnit, @naag ✌️👍💪

**Comment by naag** [comment]
Thanks, it works nicely! :-)
